pandas/core/internals/concat.py

```diff
@@ -186,6 +188,11 @@ def get_reindexed_values(self, empty_dtype, upcasted_na):
 
                 if getattr(self.block, 'is_datetimetz', False) or \
                         is_datetimetz(empty_dtype):
+                    if self.block is None:
+                        missing_arr = np.full(self.shape, fill_value)
+                        missing_time = DatetimeIndex(missing_arr[0],
```

Comment:
can u walk back up the stack and see exactly where block is None- this is a guarantee iirc of this function to not be None so this might be an error higher up

Reply:
Block is none when there are a mismatched number of columns between the two concatenating dataframes. I assumed this was correct behavior?

Reply:
when I last changed this, I was really trying NOT to use DTI directly here as we are trying isolate things like that to higher level methods.

See if you can revise, otherwise I will take a look.